src/main/java/org/rumbledb/expressions/flowr/ReturnClause.java

```diff
@@ -46,10 +46,19 @@ public Expression getReturnExpr() {
 
     @Override
     public void initHighestExecutionMode(VisitorConfig visitorConfig) {
-        this.highestExecutionMode =
-            this.previousClause.getHighestExecutionMode(visitorConfig).isDataFrame()
-                ? ExecutionMode.RDD
-                : ExecutionMode.LOCAL;
+        if (this.previousClause.getHighestExecutionMode(visitorConfig).isDataFrame()) {
+            this.highestExecutionMode = ExecutionMode.RDD;
+            return;
+        }
+        if (this.returnExpr.getHighestExecutionMode(visitorConfig).isRDD()) {
+            this.highestExecutionMode = ExecutionMode.RDD;
+            return;
+        }
+        if (this.returnExpr.getHighestExecutionMode(visitorConfig).isDataFrame()) {
+            this.highestExecutionMode = ExecutionMode.DATAFRAME;
+            return;
+        }
+        this.highestExecutionMode = ExecutionMode.LOCAL;
```

Comment:
Very nicely handled. I felt like I would be prone to making a mistake here :)

Reply:
This was not a mistake -- this was just not implemented yet (because not trivial). :)